library/bignum.c

```diff
@@ -1826,9 +1760,11 @@ int mbedtls_mpi_mod_int( mbedtls_mpi_uint *r, const mbedtls_mpi *A, mbedtls_mpi_
 }
 
 /*
- * Fast Montgomery initialization (thanks to Tom St Denis)
+ * Fast Montgomery initialization (thanks to Tom St Denis).
+ *
+ * Would be static, but we want to test it.
  */
-static void mpi_montg_init( mbedtls_mpi_uint *mm, const mbedtls_mpi *N )
+void mbedtls_mpi_montg_init( mbedtls_mpi_uint *mm, const mbedtls_mpi *N )
```

Comment:
I am wondering as to why we have an added `mbedtls_mpi` dependency while the same can be achieved way more quickly and effienctly using an input/output variable as [demonstrated in the prototype](https://github.com/hanno-arm/mbedtls/blob/ecp_prototype/library/bignum_core.c#L312-L318)

~~~~~
mbedtls_mpi_uint mbedtls_mpi_montg_init( mbedtls_mpi_uint m )
{
    mbedtls_mpi_uint m_inv = m;
    for( int i = biL; i >= 1; i /= 2 )
        m_inv *= ( 2 + ( m * m_inv ) );
    return( m_inv );
}
~~~~~

This makes it simpler to initialise on `mbedtls_mpi_mod_modulus_setup()` without having to declare and init `mbedtls_mpi`  data structures.

Reply:
happy to update during the rebase once Gabor's PR goes in

Reply:
I have moved the body of the code from the existing `mpi_montg_init()` into `bignum_new.c`, and made the old function call it

Reply:
I think the existing implementation will be faster, as its loop condition is `i >= 8` rather than `i >= 1`, and the loop contains two multiplications. These will likely take longer than the additional pre- and post-computation in the existing implementation, particularly on low-end devices.

Reply:
It is precomputation and only on a single limb: the performance impact is negligible. If it is easier to use and/or read, it is well worth the cost.